Documentation/EmacsConfiguration.md

```diff
@@ -0,0 +1,61 @@
+# Emacs Project Configuration
+
+Emacs can be configured with `lsp-mode` and `clangd` to work well.
+
+### clangd
+
+The official clangd extension can be used for C++ comprehension. You
+can use the following `.clangd` file placed in the project root:
+
+```yaml
+CompileFlags:
+  CompilationDatabase: Build/i686
+  Add:
+    - "-D__serenity__"
+    - "-UNO_TLS"
+    - "-I/path/to/serenity/Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0"
+    - "-I/path/to/serenity/Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0/i686-pc-serenity"
+```
+
+You will obviously need to update `11.2.0` to whatever your version at
+the time is.
+
+Run cmake at least once for this to work. clangd has difficulty
+finding specific methods and types, especially with inheritance
+trees. Also, include errors are wrong in 90% of cases.
+
```

Comment:
This isn't the case for me, what kind of errors are you getting?

Reply:
I actually took that verbatim from the VSCode document here: https://github.com/SerenityOS/serenity/blob/master/Documentation/VSCodeConfiguration.md?plain=1#L20

I'm happy to remove part/all of it. I _think_ you do need to run cmake at least once to generate the compile_commands.json but I don't know about the rest.